graph/resolvers/settings.js

```diff
@@ -1,3 +1,21 @@
+const {
+  VIEW_PROTECTED_SETTINGS,
+} = require('../../perms/constants');
+
+const {decorateWithPermissionCheck} = require('./util');
+
 const Settings = {};
 
+// PROTECTED_SETTINGS are the settings keys that must be protected for only some
+// eyes.
+const PROTECTED_SETTINGS = [
+  'premodLinksEnable',
+  'autoCloseStream',
+  'wordlist',
+  'domains',
+];
+
+// decorate the fields on the settings resolver with a permission check.
+decorateWithPermissionCheck(Settings, VIEW_PROTECTED_SETTINGS, ...PROTECTED_SETTINGS);
```

Comment:
Nice^^ 

My suggestion is to allow different permissions per field so that it is more flexible: 

```js
const protect = {
  premodLinksEnable: [VIEW_PROTECTED_SETTINGS]
  autoCloseStream: ...
};

decorateWithPermissionCheck(Settings, protect);
```

Reply:
Good idea, will implement.